packages/core/strapi/lib/services/entity-validator/validators.js

```diff
@@ -7,9 +7,9 @@ const { yup } = require('@strapi/utils');
 /**
  * Utility function to compose validators
  */
-const composeValidators = (...fns) => (attr, { isDraft }) => {
+const composeValidators = (...fns) => (attr, { isDraft, model, attributeName, entity, data }) => {
   return fns.reduce((validator, fn) => {
-    return fn(attr, validator, { isDraft });
+    return fn(attr, { isDraft, model, attributeName, entity, data }, validator);
```

Comment:
This change is necessary to have `composeValidators` working as expected when using in another `composeValidators`. Before it didn't pass the additional data (`isDraft`) to the next `composeValidators` which resulted in false negative/positive validations.

Eg: `emailValidator` or `uidValidator`:

Reply:
With this refactor you are still not passing down the parent validator to the nested composeValidator. 

I think we can fix this by passing down the validator to the nested compose and almost making the validator signature a bit simpler:

```js
const yup = require('yup');


const composeValidators = (...fns) => (...args) => {
  let validator = yup.mixed();

  // if we receive a schema then use it as base schema for nested composition
  if (yup.isSchema(args[0])) {
    validator = args[0]
    args = args.slice(1)
  }

  return fns.reduce((validator, fn) => fn(validator, ...args), validator);
}

const schema = composeValidators(
  () => yup.string(),
  composeValidators(
    (v, opts) => {
      return v.max(5)
    },
    (v, opts, entity) => {
      console.log(entity)
      return v.min(opts.min || 10)
    }
  )
)({ min: 2}, { id: 1})



try {
  schema.validateSync('120', { strict: true})
} catch (error) {
  console.log(error.message)
}
```